### PR TITLE
always set url parameters when they are available in the app dispatch

### LINKED ISSUE #13382: App framework passes null argument to controller when calendar app enabled
### Steps to reproduce:
1. Checkout master (2fbe8bbcba2e6387669f0028cb5a3d314336efe4)
2. Setup the calendar app from master (https://github.com/owncloud/calendar/commit/b0c5c01025ae932c38c213bb98825e9c60b47a85)
3. Setup OC
4. Upload a few files
5. Click "Favorite" on any file
6. Refresh the page

### Expected result
Favorite stays, entry properly added in `oc_vcategory_to_object` table.

### Actual result
Favorite not added in the database.
If you debug into `updateFileTags()` in the ApiController: https://github.com/owncloud/core/blob/master/apps/files/controller/apicontroller.php#L72
You'll notice that `$path` is null.
If you disable the calendar app, the `$path` will contain the correct value.
Very strange side effect!

### Versions
OC 8 master 2fbe8bbcba2e6387669f0028cb5a3d314336efe4

@Raydiation please have a look

Setting to OC 8.0 target as this could potentially break other controllers. (CC @DeepDiver1975)

### PR DESCRIPTION
**Pull request by BernhardPosselt**
If the request is accessed from any app outside of a middleware controller dispatch, the urlparameters wont be set. This PR always sets the url parameters when an app dispatch happens. Fixes the tutorial app DELETE and PUT requests for me

@LukasReschke @MorrisJobke @DeepDiver1975 @PVince81 @georgehrke 

might fix  #13382

### CODE REVIEW

**Comment by PVince81** [comment]
Just tested https://github.com/owncloud/core/issues/13382 and the fix works with the calendar app enabled!
But am not familiar enough with this part of the code to evaluate the possible impact.

**Comment by Raydiation** [comment]
@PVince81 basically only affects appframework apps. If you can delete feeds/folders in the news/app-tutorial app everything should be fine

File: lib/private/appframework/dependencyinjection/dicontainer.php [hunk 1]
```diff
@@ -233,7 +233,6 @@ public function __construct($appName, $urlParams = array()){
 			/** @var $c SimpleContainer */
 			/** @var $server SimpleContainer */
 			$server = $c->query('ServerContainer');
-			$server->registerParameter('urlParams', $c['urlParams']);
```
Comment (LukasReschke) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: This affects the public API. Applications could previously access it via `$this->request->urlParams`

Comment (LukasReschke) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: (Now it contains an empty array)

Comment (Raydiation) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: @LukasReschke should still work fine. Did you test it?

Comment (LukasReschke) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: Sure ;-)

Try a `var_dump($this->request->urlParams);` in \OCA\Files_Sharing\Controllers\ShareController::showShare

Comment (Raydiation) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: The parameters are now set here https://github.com/owncloud/core/pull/13387/files#diff-d21c280adc02d6e1c83154482ad5051eR78 which means that any app that tries to access the request gets the proper parameters

Comment (LukasReschke) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: Well, `\OCA\Files_Sharing\Controllers\ShareController::showShare` doesn't get it.

Comment (Raydiation) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: Can you give me a link how its wired up?

Comment (Raydiation) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: Or at least where the method is called. This seems like a frankenstein app, part appframework, part legacy

Comment (LukasReschke) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: Of course it's a frankenstein app! – Our favourite ;-)

Routes are defined at https://github.com/owncloud/core/blob/2fbe8bbcba2e6387669f0028cb5a3d314336efe4/core/routes.php#L98-113
Controller is at https://github.com/owncloud/core/blob/2fbe8bbcba2e6387669f0028cb5a3d314336efe4/apps/files_sharing/lib/controllers/sharecontroller.php

Comment (LukasReschke) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: (the `->urlParams` is actually never called in there but it shouldn't disappear after this patch anyways)

Comment (LukasReschke) [inline lib/private/appframework/dependencyinjection/dicontainer.php]: Resolved by https://github.com/owncloud/core/commit/f1f870127504d1f4ada8cb13831136a1c3692f50 - thanks @Raydiation!

File: lib/private/appframework/app.php [hunk 2]
```diff
@@ -75,7 +75,9 @@ public static function buildAppNamespace($appId, $topNamespace='OCA\\') {
 	 */
 	public static function main($controllerName, $methodName, DIContainer $container, array $urlParams = null) {
 		if (!is_null($urlParams)) {
-			$container['urlParams'] = $urlParams;
+			$container['OCP\\IRequest']->setUrlParameters($urlParams);
+		} else if (isset($container['urlParams']) && !is_null($container['urlParams'])){
```
Comment (LukasReschke) [inline lib/private/appframework/app.php]: Space before {

**Comment by LukasReschke** [comment]
:+1:

**Comment by LukasReschke** [comment]
@owncloud-bot Retest this please.

**Comment by MorrisJobke** [comment]
Works :+1:

**Pull request merged**